### ISSUE TITLE
[Bug]: Template + Macro rendering issue with Template Card

### ISSUE THREAD

**Issue by jeeftor** [piitaya/lovelace-mushroom#1020]
### Requirements

- [X] I have updated Mushroom to the latest available version
- [X] I did a search to see if there is a similar issue or if a pull request is open.

### Current Behavior

I have a macro + template I'm trying to use to generate a color sequence based on the state of an item:

```python
{%- macro get_mqtt_rgb(var1) -%}
{{'100,0,100' if is_state(var1,'unavailable')}}
{{'0,0,0' if is_state(var1,'off')}}{{'100,0,0' if is_state(var1,'on')}}
{%-endmacro -%}
{{get_mqtt_rgb('binary_sensor.shed_contact')}}
```

When it wasn't working I tossed it into the **Secondary Information** to see what was being rendered.

What I expect is 3 digits separated by commas:

![image](https://user-images.githubusercontent.com/6491743/218293414-eecaa740-68b0-4e37-90a1-4ceb50e0448b.png)

What i get is all the digits stuck together

If I throw something else in there: 

![image](https://user-images.githubusercontent.com/6491743/218293433-5db3162d-35fc-4f95-adcf-822fdef4ae41.png)

![image](https://user-images.githubusercontent.com/6491743/218293437-787c7521-3b7c-4a21-b7c8-0f0fa8877594.png)

The output looks odd - and doesnt match what i see in the template editor (see below)

### Expected Behavior

![image](https://user-images.githubusercontent.com/6491743/218293384-da05f096-dd67-45e2-b00e-aff14bb335c3.png)

### Context

YAML state
```yaml
unavailable
off
on
```

### Environment

```markdown
- Browser: Firefox
- HA Version: Latest Feb Release
- Mushroom: (Installed yesterday - latest realease)
```

**Comment by piitaya** [comment]
Hello 👋
Comma are interpreted as a list in jinja2. There was a similar issue: https://github.com/piitaya/lovelace-mushroom/issues/884. I think it can help you.

**Comment by jeeftor** [comment]
Funny you mention that ... I'm in the process of trying to convert things to hex this morning :) 

```
{% set state = states('sensor.door_status_payload')|from_json %}
{{ '#%02x%02x%02x' | format(state.seg.i[4][0],state.seg.i[4][1],state.seg.i[4][2]) }}
```

Giving this a go ... still not working perfectly but could be a line/feed issue where i need to toss in a `-` somewhere

(updates hopefully to follow... )

**Comment by jeeftor** [comment]
Ok! Looks like we got a win here - minus the fact I picked a terrible purple:


![image](https://user-images.githubusercontent.com/6491743/218313623-686b635e-6d76-4c68-9387-eae302bb34ee.png)

```python
{% set state = states('sensor.door_status_payload')|from_json -%}
{{ '#%02x%02x%02x' | format(state.seg.i[4][0],state.seg.i[4][1],state.seg.i[4][2]) -}}
```

If anybody stumbles on this I'm trying to sync my WLED lighting color to the same color in mushroom -> as such I wanted to generate a set of macros which i can use for either wled or mushroom.

@piitaya  - Is there somewhere this _could_ be documented example-ized in the docs? I'd be happy to draft a PR
